indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -566,8 +568,13 @@ class RHRegistrationReject(RHManageRegistrationBase):
     """Reject a registration."""
 
     def _process(self):
-        _modify_registration_status(self.registration, approve=False)
-        return jsonify_data(html=_render_registration_details(self.registration))
+        form = RejectRegistrantsForm()
+        message = _("Rejecting this registration will trigger a notification email.")
+        if form.validate_on_submit():
+            self.registration.rejection_reason = form.rejection_reason.data
+            _modify_registration_status(self.registration, approve=False)
```

Comment:
What about moving the reason update into `_modify_registration_status` (only used in case of `approve=False` of course)?